src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -409,105 +409,71 @@ public override string ReadString()
+            Read(data);
 
             long ret;
-            fixed (byte *src = &data[0]){
-                PutBytesBE ((byte *) &ret, src, 8);
-            }
-
+            SwapForBE(new Span<byte>(&ret, 8), data);
             return ret;
         }
-        public override unsafe sbyte ReadSByte()
-        {
-            byte[] data = new byte[4];
-            Read(data, 0, 4);
 
-            int ret;
-            fixed (byte *src = &data[0]){
-                PutBytesBE ((byte *) &ret, src, 4);
-            }
-            return (sbyte)ret;
-        }
-
-        public unsafe byte ReadUByte()
-        {
-            byte[] data = new byte[4];
-            Read(data, 0, 4);
-
-            int ret;
-            fixed (byte *src = &data[0]){
-                PutBytesBE ((byte *) &ret, src, 4);
-            }
-            return (byte)ret;
-        }
+        // SDB encodes these as 4 bytes
+        public override sbyte ReadSByte() => (sbyte)ReadInt32();
+        public byte ReadUByte() => (byte)ReadUInt32();
+        public ushort ReadUShort() => (ushort)ReadUInt32();
 
         public override unsafe int ReadInt32()
         {
-            byte[] data = new byte[4];
-            Read(data, 0, 4);
+            Span<byte> data = stackalloc byte[4];
+            Read(data);
+
             int ret;
-            fixed (byte *src = &data[0]){
-                PutBytesBE ((byte *) &ret, src, 4);
-            }
+            SwapForBE(new Span<byte>(&ret, 4), data);
             return ret;
         }
 
         public override unsafe double ReadDouble()
         {
-            byte[] data = new byte[8];
-            Read(data, 0, 8);
+            Span<byte> data = stackalloc byte[8];
+            Read(data);
 
             double ret;
-            fixed (byte *src = &data[0]){
-                PutBytesBE ((byte *) &ret, src, 8);
-            }
+            SwapForBE(new Span<byte>(&ret, 8), data);
             return ret;
         }
 
         public override unsafe uint ReadUInt32()
         {
-            byte[] data = new byte[4];
-            Read(data, 0, 4);
+            Span<byte> data = stackalloc byte[4];
+            Read(data);
 
             uint ret;
-            fixed (byte *src = &data[0]){
-                PutBytesBE ((byte *) &ret, src, 4);
-            }
+            SwapForBE(new Span<byte>(&ret, 4), data);
             return ret;
         }
-        public unsafe ushort ReadUShort()
-        {
-            byte[] data = new byte[4];
-            Read(data, 0, 4);
-
-            uint ret;
-            fixed (byte *src = &data[0]){
-                PutBytesBE ((byte *) &ret, src, 4);
-            }
-            return (ushort)ret;
-        }
     }
 
     internal class MonoBinaryWriter : BinaryWriter
     {
         public MonoBinaryWriter(Stream stream) : base(stream) {}
+
         public void WriteString(string val)
         {
+            var bytes = Encoding.UTF8.GetBytes(val);
```

Comment:
did you mean to use `bytes` for `Write(..)`?

Reply:
🤦